pkg/controller/datavolume/util.go

```diff
@@ -516,6 +516,36 @@ func updateDataVolumeDefaultInstancetypeLabels(client client.Client, syncState *
 		copyDefaultInstancetypeLabels(dv, pvc.Labels)
 		return nil
 	}
+	if dv.Spec.Source != nil && dv.Spec.Source.Snapshot != nil {
+		snapshot := &snapshotv1.VolumeSnapshot{}
+		key := types.NamespacedName{
+			Name:      dv.Spec.Source.Snapshot.Name,
+			Namespace: dv.Spec.Source.Snapshot.Namespace,
+		}
+		if err := client.Get(context.TODO(), key, snapshot); err != nil {
+			if k8serrors.IsNotFound(err) {
+				return nil
+			}
+			return err
+		}
+		copyDefaultInstancetypeLabels(dv, snapshot.Labels)
+		return nil
+	}
+	if dv.Spec.Source != nil && dv.Spec.Source.Registry != nil {
```

Comment:
TBH I am not sure what the registry part is about. If you're creating a new PVC with registry source, that PVC already has the needed labels per https://github.com/kubevirt/containerized-data-importer/pull/3103. Do you also want them on the DV?

Reply:
Yes that is the idea.

Reply:
I am not sure you really need them on the DV, but sure no problem